### ISSUE TITLE
NVDA reader doesn't read icon on mouse hover

### ISSUE THREAD

**Issue by Thomas5588** [nvaccess/nvda#12047]
<button role="button" class="edit" data-dismiss="modal" aria-label="Edit Employee">
                    <i class="fas fa-edit" aria-hidden="true" title="Edit"></i>
                    <span class="sr-only">Close</span>
                </button>

### System configuration
NVDA : 2020.3

**Comment by Thomas5588** [comment]
@josephsl - could you please clarify the above issue, why NVDA does not read icon aria-label  on mouse hover?

**Comment by feerrenrut** [comment]
@Thomas5588 could you please elaborate on this issue, please answer the following:
- Does the title get read when navigating with a keyboard? 
- Have you confirmed this is exposed by the browser as you expect, check the accessibility tools in Chrome, Firefox, or Edge to confirm?

**Comment by feerrenrut** [comment]
Since this snippet relies on font-awesome, I have created the following sample: https://codepen.io/reefturner/pen/PobGWrv

Mouse tracking does not seem to read the accName for the button, however NVDA reads this correctly in browse mode with the keyboard. 

I tested this with Edge `Version 88.0.705.63 (Official build) (64-bit)`

**Comment by Thomas5588** [comment]
Thanks for reply @feerrenrut 

Does the title get read when navigating with a keyboard? - NVDA read button aria-label when navigation with keyboard.
Have you confirmed this is exposed by the browser as you expect, check the accessibility tools in Chrome, Firefox, or Edge to confirm? - Could you please let me know which settings need to be check?

When I navigation with keyboard the NVDA read button's aria-label, but when I mouse hover on the button the NVDA did not read the button aria-label.

**Comment by zersiax** [comment]
Aria-label pretty much overrides whatever other label is on the button, so that bit makes sense. That the icon is not being read is probably because you set aria-hidden=true on it, although I don't have time to test, currently.

**Comment by wwwXpert** [comment]
I originally tested and reproduced the issue just with the icons themselves (no <a> tags wrapped around them)
Here's a test page with just the icons without aria-hidden="true" that can be used to test and reproduce the issue.
The screen reader fails to read the aria-label and title attributes that are present.

**Comment by tmiaa** [comment]
@Thomas5588

This should not be expected to work:

> `aria-hidden="true" title="Edit"`

because well, the element is `aria-hidden="true"`.

And side note: you don't have to use `role="button"` on `<button>`, that's the implicit role.

---

I have this case:

```html
<button aria-label="Zoom in">
  <span aria-hidden="true">+</span>
</button>
```

With or without `aria-hidden="true"` NVDA never announces <q>zoom in</q> from `aria-label` on hover.

**Comment by StefanRetief** [comment]
> I have this case:
> 
> ```
> <button aria-label="Zoom in">
>   <span aria-hidden="true">+</span>
> </button>
> ```
> 
> With or without `aria-hidden="true"` NVDA never announces zoom in from `aria-label` on hover.

@tmiaa I'm running into the same issue. Though, I have found that cases where if there is a `::before` psuedo element, the label will be read by NVDA:

```html
<button aria-label="Zoom in">
  ::before
  <span aria-hidden="true">+</span>
</button>
```

**Comment by andypillip** [comment]
I could reproduce the issue with NVDA 2022.4 in Firefox and Chrome latest.

Whether the button has any text contents does not matter, the accName will not get announced if calculated from either `aria-label` or `aria-labelledby`.

Windows Narrator correctly announces the name on hover.

For more details see: https://stackoverflow.com/questions/75064724/accessibility-aria-label-not-being-read-on-hover/75107868#75107868

**Comment by nolimitdev** [comment]
By my testing NVDA absolutely does not read title and arial-label attributes on hover. Solution is to use "invisible" text which works for tags A and BUTTON...

```html
<style>
button { padding: 0; border: none; }
.blue { background: blue; display: block; width: 32px; height: 32px; cursor: pointer; }
.orange { background: orange; display: block; width: 32px; height: 32px; cursor: pointer; }
.hidden { font-size: 1000%; color: transparent; display: block; height: 100%; width: 100%; overflow: hidden; }
</style>

<a href="" class="blue" onclick="return false;"><span class="hidden">BLUE</span></a>

<button class="orange"><span class="hidden">ORANGE</span></button>
```

**Comment by hwf1324** [comment]
I suspect this is related to `IA2TextTextInfo`, as the mouse only reports text content calculated by `TextInfo.text`.
For this issue, the value of `IA2TextTextInfo.text` seems to be the content within the `<button></button>` element.
The underlying API call should be `obj.IAccessibleTextObject.text(start, end)`.
However, the `obj.basicText` path used by `NVDAObjectTextInfo` seems to retrieve the correct content from `obj.name`.

**Comment by hwf1324** [comment]
Related: #10920

**Comment by hwf1324** [comment]
@seanbudd, I don't think this issue should be closed because its description is clearer and more precise.

**Comment by seanbudd** [comment]
Our opinion was the other issue was more clear

**Comment by hwf1324** [comment]
In my opinion, the description of issue #11845 is outdated; it lacks clear steps to reproduce the issue and actual versus expected results. Furthermore, it may be somewhat broad, and the issue isn't specific enough.
